src/Type/Php/StrlenFunctionReturnTypeExtension.php

```diff
@@ -38,19 +42,25 @@ public function getTypeFromFunctionCall(
 
 		$argType = $scope->getType($args[0]->value);
 
-		$constantScalars = TypeUtils::getConstantScalars($argType);
+		if ($argType->isSuperTypeOf(new BooleanType())->yes()) {
```

Comment:
Is this condition right? This is true for `bool` but also for `bool|object`, and also `mixed`.

Reply:
And it's false for `true` and `false`.

Reply:
Yes, this condition is right. There a test that covers it:
`assertType('int<0, 1>', strlen($emptyStringBoolNull)); // ""|bool|null `

The thing is `TypeUtils::getConstantScalars()` returns constant scalars only if the passed argument contains only constant types. So if the arg is a super type of bool, this code tries to remove bool out of it, and if `TypeUtils::getConstantScalars()` returns some constant scalars after it, then it's a case when it's a union of bool and some scalars. So in this case this code replaces bool to true|false.
If it's a `bool|object` or 'mixed', this code will do nothing, because `TypeUtils::getConstantScalars()` will return an empty array.